daemon/containerd/image_list.go

```diff
@@ -747,8 +747,8 @@ func computeSharedSize(chainIDs []digest.Digest, layers map[digest.Digest]int, s
 	return sharedSize, nil
 }
 
-// readConfig reads content pointed by the descriptor and unmarshals it into a specified output.
-func readConfig(ctx context.Context, store content.Provider, desc ocispec.Descriptor, out interface{}) error {
+// readJSON reads content pointed by the descriptor and unmarshals it into a specified output.
+func readJSON(ctx context.Context, store content.Provider, desc ocispec.Descriptor, out interface{}) error {
```

Comment:
Perhaps it could be extracted to the `github.com/containerd/containerd/content` package.

There's already [`content.ReadBlob` ](https://github.com/containerd/containerd/blob/a406da962818ba9ca8f38b4ac5215753ceb1846c/core/content/helpers.go#L84C6-L84C14), so I guess it could be named `content.ReadJSONBlob`.

Reply:
Ah, except our version also translates the `cerrdefs.NotFound` into `errdefs.NotFound`